azure-automation-plus-dsc-lab/azuredeploy.json

```diff
@@ -0,0 +1,849 @@
+      },
+      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/azure-automation-plus-dsc-lab/"
+    },
+    "_artifactsLocationSasToken": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Auto-generated token to access _artifactsLocation when the template is deployed."
+      },
+      "defaultValue": ""
+    },
+    "adminUserName": {
+      "type": "string",
+      "metadata": {
+        "description": "The administrator account of the VMs and Domain Admin."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password for the administrator account of the VMs and Domain Admin that must match default AD complexity requirements and be at least 12 characters."
+      }
+    },
+    "studentRandomInfix": {
+      "type": "string",
+      "metadata": {
+        "description": "Random infix used to construct unique resource values within the same subscription, consisting of 8 numeric and lower-case characters only."
+      },
+      "minLength": 8,
+      "maxLength": 8
+    },
+    "azureEnvironment": {
+      "type": "string",
+      "metadata": {
+        "description": "The AzureCloud (Public cloud) environment will be used."
+      },
+      "defaultValue": "AzureCloud"
+    },
+    "studentNumber": {
+      "defaultValue": 0,
+      "allowedValues": [
+        0,
+        1,
+        2,
+        3,
+        4,
+        5,
+        6,
+        7,
+        8,
+        9,
+        10,
+        11,
+        12,
+        13,
+        14,
+        15,
+        16
+      ],
+      "minValue": 0,
+      "maxValue": 16,
+      "type": "int",
+      "metadata": {
+        "description": "Enter a two digit number from 0 to 16."
+      }
+    },
+    "omsLocation": {
+      "type": "string",
+      "allowedValues": [
+        "Australia Southeast",
+        "Canada Central",
+        "Central India",
+        "East US",
+        "Japan East",
+        "Southeast Asia",
+        "UK South",
+        "West Europe"
+      ],
+      "defaultValue": "East US",
+      "metadata": {
+        "description": "Log Analytics (OMS) region."
+      }
+    }
+  },
+  "variables": {
+    "azureRegion": "[resourceGroup().location]",
+    "rgName": "[resourceGroup().name]",
+    "subId": "[subscription().id]",
+    "randomInfix": "[parameters('studentRandomInfix')]",
+    "baseUrl": "[concat(parameters('_artifactsLocation'))]",
+    "nestedBaseUrl": "[concat(variables('baseUrl'),'/nested')]",
+    "createPublicIPUri": "[concat(variables('nestedBaseUrl'),'/04.36.00.createPublicIP.json',parameters('_artifactsLocationSasToken'))]",
+    "devPublicIPAddressName": "[toLower(concat(variables('devPrefix'),'1-pip'))]",
+    "devDomainNameLabelPrefix": "[concat(variables('devPublicIPAddressName'),parameters('studentRandomInfix'))]",
+    "devPublicIPAddressType": "Static",
+    "fqdnAzureSuffix": ".cloudapp.azure.com",
```

Comment:
this needs to come from the reference() function - can't hard code it

Reply:
@bmoore-msft Which resource can I use the reference() function for to obtain the suffix before creating the public IP address?